packages/sku/src/utils/SkuConfigUpdater.ts

```diff
@@ -146,12 +159,12 @@ export class SkuConfigUpdater {
    *
    * This method does not write the changes to the file system. Use `commitConfig` to do that.
    */
-  upsertConfig<T extends keyof SkuConfig>({
+  upsertConfig<T extends AllSkuConfigKeys>({
     property,
     value,
   }: {
     property: T;
-    value: SkuConfig[T];
+    value: AllSkuConfigValues;
```

Comment:
This isn't quite the same as the previous type. `AllSkuConfigValues` is a union of all possible values of sku config properties. `SkuConfig[T]` however was the value for the specific sku config property specified in `property`.

Clearly `SkuConfig[T]` won't work anymore, but you might need to extract a separate `JoinedSkuConfig` type that is `SkuConfigBase & WebpackSkuConfig & ViteSkuConfig`, and then do `JoinedSkuConfig[T]`.

Happy to pair on this if you need.